packages/xo-web/src/common/xo/migrate-vm-modal/index.js

```diff
@@ -25,16 +23,156 @@ import { isSrShared, isSrWritable } from '../'
 
 import styles from './index.css'
 
+@connectStore(() => {
+  const getSnapshot = (_, props) => props.snapshot
+
+  const getVbds = createGetObjectsOfType('VBD')
+    .pick(createSelector(getSnapshot, snapshot => snapshot.$VBDs))
+    .filter([vbd => !vbd.is_cd_drive && vbd.VDI !== undefined])
+
+  const getVdis = createGetObjectsOfType('VDI-snapshot').pick(
+    createSelector(getVbds, vbds => mapPlus(vbds, (vbd, push) => push(vbd.VDI)))
+  )
+
+  const getVifs = createGetObjectsOfType('VIF').pick(
+    createSelector(getSnapshot, snapshot => snapshot.VIFs)
+  )
+
+  const getNetworks = createGetObjectsOfType('network')
+
+  return {
+    networks: getNetworks,
+    vdis: getVdis,
+    vifs: getVifs,
+    snapshot: getSnapshot,
+  }
+})
+class Snapshot extends BaseComponent {
+  componentWillMount () {
+    this._onChangeMapVdisSrs(this.props)
+  }
+
+  componentWillReceiveProps (props) {
+    if (props.defaultSr !== this.props.defaultSr) {
+      this._onChangeMapVdisSrs(props)
+    }
+  }
+
+  _onChange = newValues => {
+    this.props.onChange({
+      ...this.props,
+      ...newValues,
+    })
+  }
+
+  _onChangeMapVdisSrs = props => {
+    this._onChange({
+      mapVdisSrs: {
+        ...props.mapVdisSrs,
+        ...mapValues(props.vdis, () => props.defaultSr),
```

Comment:
This implies that the map you will send to xo-server in the end will contain all the `VDI(-snapshot)`s but you actually don't need to, since there is a default SR and xo-server handles the default SR correctly.

This is how it should work IMO:
- In the UI, the default SR, each VDI and each VDI-snapshot all have their `SelectSr` independant from one another (no autoselect of anything)
- When calling the API method:
  - The `sr` param is **your** `defaultSr`
  - The `mapVdisSrs` param is built as follows from **your** `mapVdisSrs`:
    - Keep the `VDI`s as they are
    - For each `VDI-snapshot`, if their `VDI`'s `SR` is not `undefined`, then set it to the same `SR`
    - Leave everything else `undefined` as they will fallback to the default SR